src/test/scala/com/fulcrumgenomics/bam/ClipOverlappingReadsTest.scala

```diff
@@ -129,6 +127,21 @@ class ClipOverlappingReadsTest extends UnitSpec {
     r1.getAlignmentEnd >= r2.getAlignmentStart shouldBe false
   }
 
+  // This is a weird test that ensures that things terminate correctly when due to highly clipped
+  // input reads, one of the reads becomes unmapped during clipping
+  it should "handle reads that get unmapped because they are fully overlapped" in {
+    val builder = new SamRecordSetBuilder(readLength=50)
+    val clipper = new ClipOverlappingReads(input=dummyBam, output=dummyBam, ref=ref)
+    val Seq(r1a, r2a) = builder.addPair(name="q1", start1=100, start2=100, cigar1="3M47S", cigar2="48S2M")
+    val Seq(r1b, r2b) = builder.addPair(name="q2", start1=100, start2=100, cigar1="2M48S", cigar2="47S3M")
+
+    for ((r1, r2) <- Seq((r1a, r2a), (r1b, r2b))) {
+      clipper.clip(r1, r2)
+      val ok = r1.getReadUnmappedFlag || r2.getReadUnmappedFlag || r1.getAlignmentEnd < r2.getAlignmentStart
+      ok shouldBe true
+    }
```

Comment:
The test states that there will be a read that is unmapped.  If the records get incorrectly modified by the clipping utility such that `r1a.end < r2a.start` and `r1b.end < r2b.start` then this test would still pass!